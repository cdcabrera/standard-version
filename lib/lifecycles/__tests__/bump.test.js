const bump = require('../bump');

describe('bump', () => {
  const mockInfo = jest.fn();
  let spy;

  beforeEach(() => {
    spy = jest.spyOn(console, 'info').mockImplementation(mockInfo);
  });

  afterEach(() => {
    mockInfo.mockClear();
    spy.mockClear();
  });

  it('should skip returning version information', async () => {
    const bumpSkip = await bump({ skip: { bump: true } }, 'lorem ipsum');

    expect({
      bumpSkip
    }).toMatchSnapshot('results');
    expect(mockInfo.mock.calls).toMatchSnapshot('console');
  });

  it('should run lifecycles returning version information', async () => {
    const bumpLifecycle = await bump(
      {
        bumpFiles: [],
        skip: {},
        scripts: {
          prerelease: 'echo "prerelease lifecycle mock"',
          prebump: 'echo "prebump lifecycle mock"',
          postbump: 'echo "postbump lifecycle mock"'
        }
      },
      'lorem ipsum'
    );

    expect({
      bumpLifecycle
    }).toMatchSnapshot('results');
    expect(mockInfo.mock.calls).toMatchSnapshot('console');
  });

  it('should run release-as returning version information', async () => {
    const bumpReleaseAs = await bump(
      {
        bumpFiles: [],
        releaseAs: '1.5.0',
        skip: {},
        scripts: {
          prerelease: 'echo "prerelease releaseAs mock"'
        }
      },
      '1.0.0'
    );

    const bumpReleaseAsStdout = await bump(
      {
        bumpFiles: [],
        skip: {},
        scripts: {
          prerelease: 'echo "prerelease releaseAsStdout mock"',
          prebump: 'echo "2.0.0"'
        }
      },
      '1.0.0'
    );

    const bumpRelease = await bump(
      {
        bumpFiles: [],
        skip: {},
        scripts: {
          prerelease: 'echo "prerelease release mock"'
        }
      },
      '0.0.0'
    );

    const bumpReleasePatch = await bump(
      {
        bumpFiles: [],
        releaseAs: 'patch',
        skip: {},
        scripts: {
          prerelease: 'echo "prerelease release patch mock"'
        }
      },
      '0.0.0'
    );

    const bumpReleaseMinor = await bump(
      {
        bumpFiles: [],
        releaseAs: 'minor',
        skip: {},
        scripts: {
          prerelease: 'echo "prerelease release minor mock"'
        }
      },
      '0.0.0'
    );

    const bumpReleaseMajor = await bump(
      {
        bumpFiles: [],
        releaseAs: 'major',
        skip: {},
        scripts: {
          prerelease: 'echo "prerelease release major mock"'
        }
      },
      '0.0.0'
    );

    expect({
      bumpReleaseAs,
      bumpReleaseAsStdout,
      bumpRelease,
      bumpReleasePatch,
      bumpReleaseMinor,
      bumpReleaseMajor
    }).toMatchSnapshot('results');

    expect(mockInfo.mock.calls).toMatchSnapshot('console');
  });

  it('should run first release returning version information', async () => {
    const bumpReleaseFirst = await bump(
      {
        bumpFiles: [],
        firstRelease: true,
        skip: {},
        scripts: {
          prerelease: 'echo "prerelease first release mock"'
        }
      },
      '0.1.0'
    );

    expect({
      bumpReleaseFirst
    }).toMatchSnapshot('results');

    expect(mockInfo.mock.calls).toMatchSnapshot('console');
  });
});
