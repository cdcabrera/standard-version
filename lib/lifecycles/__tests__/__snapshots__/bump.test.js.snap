// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bump should run first release returning version information: console 1`] = `
Array [
  Array [
    "<green>✔</green> Running lifecycle script \\"<bold>prerelease</bold>\\"",
  ],
  Array [
    "<blue>ℹ</blue> - execute command: \\"<bold>echo \\"prerelease first release mock\\"</bold>\\"",
  ],
  Array [
    "<red>✖</red> skip version bump on first release",
  ],
]
`;

exports[`bump should run first release returning version information: results 1`] = `
Object {
  "bumpReleaseFirst": "0.1.0",
}
`;

exports[`bump should run lifecycles returning version information: console 1`] = `
Array [
  Array [
    "<green>✔</green> Running lifecycle script \\"<bold>prerelease</bold>\\"",
  ],
  Array [
    "<blue>ℹ</blue> - execute command: \\"<bold>echo \\"prerelease lifecycle mock\\"</bold>\\"",
  ],
  Array [
    "<green>✔</green> Running lifecycle script \\"<bold>prebump</bold>\\"",
  ],
  Array [
    "<blue>ℹ</blue> - execute command: \\"<bold>echo \\"prebump lifecycle mock\\"</bold>\\"",
  ],
  Array [
    "<green>✔</green> Running lifecycle script \\"<bold>postbump</bold>\\"",
  ],
  Array [
    "<blue>ℹ</blue> - execute command: \\"<bold>echo \\"postbump lifecycle mock\\"</bold>\\"",
  ],
]
`;

exports[`bump should run lifecycles returning version information: results 1`] = `
Object {
  "bumpLifecycle": null,
}
`;

exports[`bump should run release-as returning version information: console 1`] = `
Array [
  Array [
    "<green>✔</green> Running lifecycle script \\"<bold>prerelease</bold>\\"",
  ],
  Array [
    "<blue>ℹ</blue> - execute command: \\"<bold>echo \\"prerelease releaseAs mock\\"</bold>\\"",
  ],
  Array [
    "<green>✔</green> Running lifecycle script \\"<bold>prerelease</bold>\\"",
  ],
  Array [
    "<blue>ℹ</blue> - execute command: \\"<bold>echo \\"prerelease releaseAsStdout mock\\"</bold>\\"",
  ],
  Array [
    "<green>✔</green> Running lifecycle script \\"<bold>prebump</bold>\\"",
  ],
  Array [
    "<blue>ℹ</blue> - execute command: \\"<bold>echo \\"2.0.0\\"</bold>\\"",
  ],
  Array [
    "<green>✔</green> Running lifecycle script \\"<bold>prerelease</bold>\\"",
  ],
  Array [
    "<blue>ℹ</blue> - execute command: \\"<bold>echo \\"prerelease release mock\\"</bold>\\"",
  ],
  Array [
    "<green>✔</green> Running lifecycle script \\"<bold>prerelease</bold>\\"",
  ],
  Array [
    "<blue>ℹ</blue> - execute command: \\"<bold>echo \\"prerelease release patch mock\\"</bold>\\"",
  ],
  Array [
    "<green>✔</green> Running lifecycle script \\"<bold>prerelease</bold>\\"",
  ],
  Array [
    "<blue>ℹ</blue> - execute command: \\"<bold>echo \\"prerelease release minor mock\\"</bold>\\"",
  ],
  Array [
    "<green>✔</green> Running lifecycle script \\"<bold>prerelease</bold>\\"",
  ],
  Array [
    "<blue>ℹ</blue> - execute command: \\"<bold>echo \\"prerelease release major mock\\"</bold>\\"",
  ],
]
`;

exports[`bump should run release-as returning version information: results 1`] = `
Object {
  "bumpRelease": "0.0.1",
  "bumpReleaseAs": "1.5.0",
  "bumpReleaseAsStdout": "2.0.0",
  "bumpReleaseMajor": "1.0.0",
  "bumpReleaseMinor": "0.1.0",
  "bumpReleasePatch": "0.0.1",
}
`;

exports[`bump should skip returning version information: console 1`] = `Array []`;

exports[`bump should skip returning version information: results 1`] = `
Object {
  "bumpSkip": "lorem ipsum",
}
`;
